cvat/apps/engine/serializers.py

```diff
@@ -1196,12 +1196,29 @@ def update(self, instance, validated_data):
         _update_related_storages(instance, validated_data)
 
         instance.save()
+
+        if 'label_set' in validated_data and not instance.project_id:
+            self.update_children_objects_on_labels_update(instance)
+
         instance.task_labels_count = instance.label_set.filter(
             parent__isnull=True).count()
         instance.proj_labels_count = instance.project.label_set.filter(
             parent__isnull=True).count() if instance.project else 0
         return instance
 
+    def update_children_objects_on_labels_update(self, instance: models.Task):
```

Comment:
```suggestion
    def update_children_on_labels_update(self, instance: models.Task):
```

`child_objects` would also be okay.